internal/graphicsdriver/opengl/gl/procaddr_darwin.go

```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
+
+package gl
+
+import (
+	"github.com/ebiten/purego"
+)
+
+func getProcAddress(namea string) uintptr {
+	return purego.Dlsym(purego.RTLD_DEFAULT, namea)
```

Comment:
Does this really work without specifying `-framework OpenGL`? Don't we need a compiler directive to load OpenGL.framework?

Reply:
I’ve ran all the tests and it functions properly. I could add a call to Dlopen in an init function for security but it doesn’t appear necessary

Reply:
I believe the compiler directive only works in cgo. We’d have to do something like `var _opengl = purego.Dlopen("OpenGL.framework", purego.RTLD_GLOBAL)`. And then we can use `_opengl` in `Dlsym` which should improve the speed of finding the symbols. I think we should do it now for this reason alone

Reply:
I thought `//go:cgo_import_dynamic` would work (e.g. https://cs.opensource.google/go/go/+/master:src/crypto/x509/internal/macos/security.go;l=82) but this seems a different thing.

Reply:
You’re correct that that would work! I didn’t think you would want to use more compiler magic comments. I’ll add now then.

Reply:
Ah dlopen might be faster? Hmmm then dlopen might be sightly better. Sorry for going back and forth.

Reply:
According to Apple’s [docs](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/dlsym.3.html): “ If dlsym() is called with the special handle RTLD_DEFAULT […] This can be a costly search and should be avoided.” Although I’m not sure how much of a difference it’ll make.

Reply:
Thank you for the link. Let's go with dlopen if possible.

Reply:
Sure thing! I have everything ready. I just need [purego#2](https://github.com/ebiten/purego/pull/2) to be merged.

Reply:
Merged :-)